smaps/src/main/java/com/google/smaps/Region.java

```diff
@@ -0,0 +1,90 @@
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.smaps;
+
+import com.google.auto.value.AutoValue;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Represents a region of memory from the dump.
+ */
+@AutoValue
+abstract class Region {
+  private String startLoc;
+  private String endLoc;
+  private String permissions;
+  private long size;
+  private long kernelPageSize;
+  private long mmuPageSize;
+  private long rss;
+  private long pss;
+  private long sharedClean;
+  private long sharedDirty;
+  private long privateClean;
+  private long privateDirty;
+  private long referenced;
+  private long anonymous;
+  private long lazyFree;
+  private long anonHugePages;
+  private long shmemHugePages;
+  private long shmemPmdMapped;
+  private long sharedHugetlb;
+  private long privateHugetlb;
+  private long hugePFNMap;
+  private long swap;
+  private long swapPss;
+  private long locked;
+  private List<String> vmFlags;
```

Comment:
You don't need any of these (line 27-51). The ones listed below cover it for you with AutoValue.